src/librustc_trans/context.rs

```diff
@@ -967,6 +967,12 @@ impl<'b, 'tcx> CrateContext<'b, 'tcx> {
         name.push_str(prefix);
         name.push_str(".");
         base_n::push_str(idx as u64, base_n::MAX_BASE, &mut name);
+
+        // MSP430 assembler does not like '@' character in symbol names.
+        if self.sess().target.target.arch == "msp430" {
+            name = str::replace(&name[..], "@", "_");
```

Comment:
Aaaah. What is even `MAX_BASE`? FWIW `@` is one of the characters we "escape" in symbols, on all targets.
So we don't actually consider it safe enough to use, although here it doesn't matter as much because it's unexported.

Another question nagging me is: do we really need this function? LLVM can handle multiple definitions with no name or with the same "informative" name without us needing to keep our own counters. Where are we keeping these `String`s generated here, and if they're stored in `CrateContext`, why isn't the `ValueRef` kept instead?